diff --git a/dist/plugins/open-api/index.mjs b/dist/plugins/open-api/index.mjs
index 8d65eb4693e8e5ab7ad99aaf2fb7e0afb8c2103f..a5e0762888ef1c733e0c4e70b76a177d94243d9f 100644
--- a/dist/plugins/open-api/index.mjs
+++ b/dist/plugins/open-api/index.mjs
@@ -43,6 +43,8 @@ function getTypeFromZodType(zodType) {
     return "object";
   } else if (zodType instanceof ZodArray) {
     return "array";
+  } else if (zodType instanceof ZodOptional) {
+    return getTypeFromZodType(zodType.unwrap());
   }
   return "string";
 }
@@ -56,6 +58,9 @@ function getFieldSchema(field) {
   if (field.input === false) {
     schema.readOnly = true;
   }
+  if (field.enum) {
+    schema.enum = field.enum;
+  }
   return schema;
 }
 function getParameters(options) {
